middlewares/team.middleware.js

```diff
@@ -119,6 +120,65 @@ async function ensureSpace(req, res, next) {
     return next();
 }
 
+async function updateTeam(req, res, next) {
+    const team = await Services.Team.updateOne(req.body.team._id, req.body.teamDetails);
+
+    if (!team) {
+        return next({
+            status: 404,
+            message: Constants.Error.TEAM_404_MESSAGE,
+            data: req.body.teamId
+        });
+    }
+
+    req.body.team = team;
+    return next();
+}
+
+async function getByHackerId(req, res, next) {
+    // might be in req.params.hackerId
+    const hacker = await Services.Hacker.findById(req.body.hackerId);
+
+    if (!hacker) {
+        return next({
+            status: 404,
+            message: Constants.Error.HACKER_404_MESSAGE,
+            data: req.body.hackerId
+        });
+    }
+
+    const team = await Services.Team.findById(hacker.teamId);
+
+    if (!team) {
+        return next({
+            status: 404,
```

Comment:
We should make this a `500` error, because a hacker should never have a `teamId` that does not point to a team.